### PR TITLE
Collect back comments

### LINKED ISSUE #107: Comment parsing bug: incomplete parsing of comments
A comment containing whitespace will not be parsed completely.

```rust
fn main() {
    // this is an example ed25519 key generated for this issue with
    //   ssh-keygen -t ed25519 -a 100 -f issue_example -C "jrimbault's stuff"
    let key = "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIGuu/zz89yZCqYOZ1Nrj/eSrLqLZyqg9iZL4iSntAOzf jrimbault's stuff";
    let key = openssh_keys::PublicKey::parse(key).unwrap();
    assert_eq!(key.comment, Some("jrimbault's"));
}
```

Is this intended behavior ?

It seems like this is coming from this section https://github.com/coreos/openssh-keys/blob/f82d3a6aba908b79bd5e649657f1d00c0eae9f59/src/lib.rs#L265-L278

I think either the remaining parts should be re-concatenated together, or you should do do 2 [`str::splice_once`](https://doc.rust-lang.org/stable/std/primitive.str.html#method.split_once) instead of 1 unconditional [`str::split_whitespace`](https://doc.rust-lang.org/stable/std/primitive.str.html#method.split_whitespace).

`split_whitepace` isn't appropriate for parsing in my opinion since it loses information.

Relevant RFC section: [RFC 4253 §4.2](https://datatracker.ietf.org/doc/html/rfc4253#section-4.2), notice the `comments` plural.

### PR DESCRIPTION
**Pull request by jRimbault**
<details>
<summary>Previous comment</summary>

This should eventually be changed to an alloc-less version but this already doesn't allocate on a commentless key

I tried with:

```rust
let (keytype, remaining) = key
   .split_once(char::is_whitespace)
   .ok_or(OpenSSHKeyError::InvalidFormat)?; 
let (data, comment) = remaining
   .split_once(char::is_whitespace)
   .ok_or(OpenSSHKeyError::InvalidFormat)?;
let comment = if comment.is_empty() {
   None
} else {
   Some(comment.to_owned())
};
```

But some unit tests didn't pass, so I'm leaving it here.

</details>

Should fix #107 

This shouldn't be a breaking change.

### CODE REVIEW

**Comment by jlebon** [comment]
This seems sane to me. I can confirm indeed that openssh supports comment fields with spaces or tabs.

The code as written doesn't preserve arbitrary space amounts, or tabs. The alternate version looks more correct. What unit tests failed with it?

**Comment by jRimbault** [comment]
Here are the 2 tests which fail, `hostname_parse` and `read_keys`:

```
failures:

---- tests::hostname_parse stdout ----
thread 'tests::hostname_parse' panicked at src/lib.rs:939:264:
called `Result::unwrap()` on an `Err` value: InvalidBase64 { detail: "Invalid byte 45, offset 5." }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- tests::read_keys stdout ----
thread 'tests::read_keys' panicked at src/lib.rs:980:69:
called `Result::unwrap()` on an `Err` value: InvalidFormat


failures:
    tests::hostname_parse
    tests::read_keys
```

I agree that the `split_once` version _seems_ more correct, since it actually preserves the _kind_ of whitespace in the comment. But since tests failed I assumed something was wrong with it.

If you want to test it locally, I committed it to my repo in the branch `split_once`.

**Comment by jRimbault** [comment]
I found the issue. When `data` didn't have a trailing space, `split_once` returned `None` instead of something like `Some(data, <emptystring>)`.

File: src/lib.rs [hunk 1]
```diff
@@ -263,19 +263,33 @@ impl PublicKey {
     }
 
     fn try_key_parse(key: &str) -> Result<Self> {
+        fn split_once_with_remaining(text: &str) -> Option<(&str, Option<&str>)> {
+            if text.contains(char::is_whitespace) {
+                text.split_once(char::is_whitespace)
+                    .map(|(s, r)| (s, Some(r)))
+            } else {
+                Some((text, None))
+            }
```
[the remark this answers is not in the document]
Comment (jRimbault) [reply src/lib.rs:272]: You're right 👍

**Comment by jRimbault** [comment]
I applied your suggestion, tests seem to be ok :)

File: src/lib.rs [hunk 2]
```diff
@@ -611,7 +616,7 @@ impl PublicKey {
     /// `fb:a0:5b:a0:21:01:47:33:3b:8d:9e:14:1a:4c:db:6d` .
     pub fn fingerprint_md5(&self) -> String {
         let mut sh = Md5::default();
-        sh.update(&self.data());
+        sh.update(self.data());
```
Comment (jlebon) [inline src/lib.rs:619]: This looks like an unrelated clippy fix. Let's keep that as a separate commit. Everything else, we can squash to one commit.

**Review by jlebon — COMMENTED** [review]
Thanks, LGTM overall. Can you also add an entry in the release notes?

**Comment by jRimbault** [comment]
Tis done!

**Comment by jRimbault** [comment]
Hi, is there an ETA on this? :)

**Review by jlebon — APPROVED** [review]

**Comment by jRimbault** [comment]
@jlebon
